Cython/Compiler/ModuleNode.py

```diff
@@ -2685,7 +2685,15 @@ def generate_module_state_traverse(self, env, code):
             Naming.fusedfunction_type_cname)
         code.putln('#endif')
 
-    def generate_module_init_func(self, imported_modules, env, code):
+    def generate_cached_methods_initializers(self, globalstate, code):
+        code.putln("#if CYTHON_COMPILING_IN_LIMITED_API")
+        for (type_cname, method_name), cname in sorted(globalstate.cached_cmethods.items()):
+            method_name_cname = globalstate.get_interned_identifier(EncodedString(method_name)).cname
```

Comment:
I initially tried to do it all in Code.py but found that at that stage in compilation nothing had yet been written to the `init_module` part of globalstate. So I moved the relevant bits here :|

Reply:
I meant only the call to `EncodedString()`. `method_name` should already be an `EncodedString` at this point and not need to become one.

Reply:
It looks like it is not an `EncodedString`, as I get a missing attribute `is_unicode` on compilation (it's a `str`)

Reply:
Yes, but, as I wrote above:
> Instead of calling EncodedString() here and in the declaration generation, we should do it in Code.py, inject_unbound_methods().